### ISSUE TITLE
considerd to add albert?

### ISSUE THREAD

**Issue by fengzuo97** [huggingface/transformers#1370]

**Comment by gooofy** [comment]
Would definitely love to see an implementation of ALBERT added to this repository. Just for completeness:

* paper: https://arxiv.org/abs/1909.11942
* reddit: https://www.reddit.com/r/MachineLearning/comments/d9tdfo/albert_a_lite_bert_for_selfsupervised_learning_of/
* medium: https://medium.com/syncedreview/googles-albert-is-a-leaner-bert-achieves-sota-on-3-nlp-benchmarks-f64466dd583

That said, it could be even more interesting to implement the core improvements (factorized embedding parameterization, cross-layer parameter sharing)  from ALBERT in (some?/all?)  other transformers as optional features?

**Comment by BramVanroy** [comment]
Knowing how fast the team works, I would expect ALBERT to be implemented quite soon. That being said, I haven't had time to read the ALBERT paper yet, so it might be more difficult than previous BERT iterations such as distilbert and RoBERTa.

**Comment by illcat** [comment]
I think ALBERT is very cool! Expect...

**Comment by ahotrod** [comment]
No code released yet by the paper's authors that I can find.

There's this implementation by others:  [https://github.com/brightmart/albert_zh](url)

**Comment by wassname** [comment]
And in pytorch https://github.com/lonePatient/albert_pytorch

**Comment by ahotrod** [comment]
And another:  [https://github.com/graykode/ALBERT-Pytorch](url)

**Comment by sarim-zafar** [comment]
Any Update on the progress?

**Comment by BramVanroy** [comment]
The paper will be presented at ICLR in April 2020. From what I last heard, the huggingface team has been talking with the people over at Google AI to share the details of the model, but I can imagine that the researchers rather wait until the paper has been presented. One of those reasons being that they want to get citations from their ICLR talk rather than an arXiv citation which, in the field, is "worth less" than a big conference proceeding. 

For now, just be patient. I am sure that the huggingface team will have a big announcement (follow their Twitter/LinkedIn channels) with a new version bump. No need to keep bumping this topic.

**Comment by roccqqck** [comment]
https://github.com/interviewBubble/Google-ALBERT

**Comment by tholor** [comment]
The official code and models got released :slightly_smiling_face: 
https://github.com/google-research/google-research/tree/master/albert

**Comment by kamalkraj** [comment]
[WIP]
ALBERT in tensorflow 2.0
https://github.com/kamalkraj/ALBERT-TF2.0

**Comment by lonePatient** [comment]
https://github.com/lonePatient/albert_pytorch

Dataset: MNLI
Model: ALBERT_BASE_V2
Dev accuracy : 0.8418

Dataset: SST-2
Model: ALBERT_BASE_V2
Dev accuracy :0.926

**Comment by stefan-it** [comment]
PR was created, see here:

https://github.com/huggingface/transformers/pull/1683

**Comment by BramVanroy** [comment]
I'm not sure how heavily you want to take advantage of Apex when available, but Apex does provide a fused implementation of Lamb. https://nvidia.github.io/apex/optimizers.html#apex.optimizers.FusedLAMB

**Comment by kamalkraj** [comment]
> [WIP]
> ALBERT in tensorflow 2.0
> https://github.com/kamalkraj/ALBERT-TF2.0

Verison 2 weights added.
Support for SQuAD 1.1 and 2.0 added. 
Reproduces the same results from paper. From my experiments, ALBERT model is very sensitive to hyperparameter like Batch Size. FineTuning using AdamW as Default in Original Repo. AdamW performs better than LAMB on Model finetuning.
